web/src/user_profile.ts

```diff
@@ -1168,7 +1168,22 @@ export function show_edit_user_info_modal(user_id: number, $container: JQuery):
         const user_id = Number($("#edit-user-form").attr("data-user-id"));
         function handle_confirm(): void {
             const url = "/json/users/" + encodeURIComponent(user_id);
-            dialog_widget.submit_api_request(channel.del, url, {});
+            const is_spammer = $("#is-user-spammer").is(":checked");
+            const message_delete_action = {
+                delete_public_stream_messages: $("#message-delete-action[value='1']").is(
```

Comment:
We should avoid using hardcoded values like 1,2 here. You can use a loop over `settings_config.message_delete_action_values` and use `key` and `code` fields to create the `message_delete_action` object.